src/platform/plugins/shared/navigation/public/top_nav_menu/top_nav_menu_item.tsx

```diff
@@ -68,9 +68,15 @@ export function TopNavMenuItem(props: TopNavMenuItemProps) {
     }
   }
 
-  function handleClick(e: MouseEvent<HTMLButtonElement>) {
-    if (isDisabled()) return;
-    props.run(e.currentTarget);
+  const isModifiedEvent = (event: MouseEvent) =>
+    !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey);
+
+  function handleClick(event: MouseEvent<HTMLButtonElement | HTMLAnchorElement>) {
+    if (isDisabled() || isModifiedEvent(event)) return;
```

Comment:
I think we should return on `isModifiedEvent(event)` only if this was a link click.  I don't think we should block onClick() for buttons. 

so something like this: 

```
if (isDisabled()) return;
if (prop.href && isModifiedEvent(event)) return; 
```

Reply:
Definitely, thanks for the catch!

Reply:
Good catch! On a side note, I just learned from another review yesterday that we now have a util specifically for this called [`getRouterLinkProps`](https://github.com/elastic/kibana/blob/c1d055973877bc66eec5e2cbf6b3b1a53396fdf5/src/platform/packages/shared/kbn-router-utils/src/get_router_link_props/index.ts) (it also handles some other edge cases). Not something for this PR, but it's worth considering if `top_nav_menu_item` should adopt this too for items with `href`.